tests/generation/test_utils.py

```diff
@@ -98,9 +98,16 @@ class GenerationTesterMixin:
 
     def _get_input_ids_and_config(self, batch_size=2):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
-        input_ids = inputs_dict[self.input_name]
+        input_ids = inputs_dict.pop(self.input_name)[:batch_size, :]
+        _ = inputs_dict.pop("attention_mask", None)
+        inputs_dict = {k: v[:batch_size, ...] if "head_mask" not in k else v for k, v in inputs_dict.items()}
 
-        input_ids = input_ids[:batch_size]
+        # we don't want encoder-decoder models to start from filled decoder ids
+        _ = inputs_dict.pop("decoder_input_ids", None)
+        _ = inputs_dict.pop("decoder_attention_mask", None)
+
+        # we'll set cache use in each test differently
+        _ = inputs_dict.pop("use_cache", None)
```

Comment:
Hope it's okay this way, otherwise we can override this in each model which returns `decoder inputs`. I don't think we need a new helper like `get_input_for_generation` as the current one works good already